ui/job-view/redux/stores/pushes.js

```diff
@@ -43,6 +43,36 @@ const getRevisionTips = (pushList) => {
   };
 };
 
+const getBugIds = (results) => {
+  const bugIds = new Set();
+
+  results.forEach((result) => {
+    const { revisions } = result;
+
+    revisions.forEach((revision) => {
+      const comment = revision.comments.split('\n')[0];
+      const bugMatches = comment.match(/-- ([0-9]+)|bug.([0-9]+)/gi);
+      if (bugMatches)
+        bugMatches.forEach((bugMatch) => bugIds.add(bugMatch.split(' ')[1]));
+    });
+  });
+  return bugIds;
+};
+
+const getBugSummaryMap = async (bugIds, dispatch) => {
+  const bugNumbers = [...bugIds];
+  const { data } = await getData(bugzillaBugsApi('bug', { id: bugNumbers }));
+  const bugData = data.bugs.reduce((accumulator, curBug) => {
+    accumulator[curBug.id] = curBug.summary;
+    return accumulator;
+  }, {});
+
+  dispatch({
```

Comment:
I was testing this today, and I noticed that this will overwrite the previous set of bug summaries (if you fetch more pushes).  So I think the best approach for this will be to use a `thunk` like some of the other functions.  You can see an example of this in `fetchPushes`.  Otherwise, you could have `addPushes` take another param of the existing `bugSummaryMap`.  Then `dispatch` the new `ADD_BUG_SUMMARIES` event with the old set as well as the new additions.

Also, I had missed this part in my earlier reviews.  We should create a new constant for this action called `ADD_BUG_SUMMARIES` instead of reusing `ADD_PUSHES` here.  And we'll need to add handling for it down in the `reducer`.

Reply:
Chose option 2's approach.

Reply:
Great!